File: apis/src/main/java/com/beat/apis/performance/application/dto/performanceDetail/PerformanceDetailCastResponse.java
package com.beat.apis.performance.application.dto.performanceDetail;

import com.beat.global.support.jackson.CdnImageUrl;

public record PerformanceDetailCastResponse(
	Long castId,
	String castName,
	String castRole,
	@CdnImageUrl String castPhoto
) {
	public static PerformanceDetailCastResponse of(Long castId, String castName, String castRole, String castPhoto) {
		return new PerformanceDetailCastResponse(castId, castName, castRole, castPhoto);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/PerformanceDetailCastResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.CastResult
import com.beat.global.support.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class PerformanceDetailCastResponse private constructor(
    val castId: Long?,
    val castName: String?,
    val castRole: String?,
    @field:CdnImageUrl val castPhoto: String?,
) {
    companion object {
        fun from(result: CastResult): PerformanceDetailCastResponse =
            PerformanceDetailCastResponse(result.id, result.name, result.role, result.photo)
    }
}
